src/mosquito/queue.cr

```diff
@@ -50,9 +36,12 @@ module Mosquito
       Redis.instance.zadd scheduled_q, execute_time.epoch_ms, task.id
     end
 
-    def dequeue
+    def dequeue : Task?
       return if empty?
+      q_config = Redis.instance.retrieve_hash config_q
+      return if q_config["next_batch"].to_i64 > Time.utc_now.epoch
       if task_id = Redis.instance.rpoplpush waiting_q, pending_q
+        Redis.instance.hset config_q, "executed", 0 if Time.utc_now.epoch > (Time.epoch(q_config["last_executed"].to_i64) + q_config["period"].to_i.seconds).epoch
```

Comment:
At first glance, I think this logic might live better in the Runner class. Thoughts?

Reply:
I could move it to before the task gets executed?

Reply:
I remember now.  As far as i can tell, I dont have access to the queue from the `Task` object?  

`undefined method 'queue' for Mosquito::NilJob (compile-time type is Mosquito::Job+)`

Because the job on the task in runner is `NilJob`?

Reply:
Ah, yeah I see. I think the reason I was saying that is that it feels a little like scope creep to have this class be writing to redis, but I'm not sure there's a way around it at the moment. 👍